cpp/src/io/orc/stripe_data.cu

```diff
@@ -1787,13 +1789,17 @@ __global__ void __launch_bounds__(block_size)
                 seconds += get_gmt_offset(tz_table.ttimes, tz_table.offsets, seconds);
               }
               if (seconds < 0 && nanos != 0) { seconds -= 1; }
-              if (s->chunk.ts_clock_rate)
+              if (s->chunk.ts_clock_rate) {
+                duration_ns d_ns{nanos};
+                d_ns += duration_s{seconds};
```

Comment:
`nanos` and `seconds` should be constructed as `duration_ns` and `duration_s` to begin with.

Reply:
This is possible for `seconds` but seconds will do some manual timezone operations by invoking `get_gmt_offset()`. My plan was to leave it for the timezone refactoring. As for `nanos`, there are bitwise operations like "shift" and "and" which are not supported by duration type.